src/ripple/app/tx/impl/Transactor.h

```diff
@@ -98,32 +94,29 @@ class Transactor
     explicit
     Transactor (ApplyContext& ctx);
 
-    void calculateFee ();
-
-    // VFALCO This is the equivalent of dynamic_cast
-    //        to discover the type of the derived class,
-    //        and therefore bad.
-    virtual
-    bool
-    mustHaveValidAccount()
-    {
-        return true;
-    }
-
     // Returns the fee, not scaled for load (Should be in fee units. FIXME)
     virtual std::uint64_t calculateBaseFee ();
 
     virtual void preCompute();
-    virtual TER checkSeq ();
-    virtual TER payFee ();
-    virtual TER checkSign ();
+
     virtual TER doApply () = 0;
 
 private:
+    TER checkSeq ();
+    TER checkSign ();
+    TER payFee ();
     TER checkSingleSign ();
     TER checkMultiSign ();
 };
 
+/** Performs early sanity checks on the account and fee fields */
+TER
+preflight1 (PreflightContext const& ctx);
+
+/** Checks whether the signature appears valid */
+TER
+preflight2 (PreflightContext const& ctx);
+
```

Comment:
@ximinez I was actually thinking of eliminating `preflight2` completely by moving it in the calling context, since every transactor (other than `Change`, which we can work around) needs it. This would guarantee that it always happens and eliminate the need for every method to call `preflight2` at the end.